kiegroup/appformer
# AF-2766: Dashbuilder Prometheus data-set provider

PR for https://issues.redhat.com/browse/AF-2766

This PR introduces a new Data Set provider for Prometheus[1].

It works by sending requests to the http API[2]  and then transforming the response in a the data set object, which is understood by Dashbuilder and can be used to build dashboards.

### Testing

* Start Prometheus. You can download a distribution in [1] or use the docker image;
* Start Prometheus with the default settings;
* In Business Central go to Data Sets and select Prometheus:

![image](https://user-images.githubusercontent.com/359820/104254112-e02d8e00-5454-11eb-97f2-a9eb65a21a6e.png)

* In the dataset edit page you can enter any query. I particularly like the "prometheus_http_requests_total" metric:

![image](https://user-images.githubusercontent.com/359820/104254190-06ebc480-5455-11eb-8e41-98b2c2ff534e.png)

* Test and save the dataset. Now you should be able to use it in a page. Here's an example:

![image](https://user-images.githubusercontent.com/359820/104254239-2da9fb00-5455-11eb-9b32-9fac1e4a7f9b.png)

For credentials set the following system properties: `dashbuilder.prometheus.user` and `dashbuilder.prometheus.password`

[1] http://prometheus.io/
[2] https://prometheus.io/docs/prometheus/latest/querying/api/


Part of an ensemble. Merge with:

kiegroup/droolsjbpm-build-bootstrap#1566
https://github.com/kiegroup/kie-soup/pull/172
